real_time/arduino_src/radio_buggy_mega/main.cpp

```diff
@@ -418,30 +388,33 @@ int main(void)
         unsigned long g_encoder_ticks_safe = g_encoder_ticks;
         sei(); //enable interrupts
 
-
+        // check for RC timout first
         if(delta1 > CONNECTION_TIMEOUT_US ||
            delta2 > CONNECTION_TIMEOUT_US ||
            delta3 > CONNECTION_TIMEOUT_US)
         {
             // we haven't heard from the RC receiver in too long
             dbg_printf("Timed out connection from RC!\n");
-            if(g_brake_needs_reset == false) 
-            {
-                g_rbsm.Send(RBSM_MID_ERROR, RBSM_EID_RC_LOST_SIGNAL);
-            }
-            g_brake_needs_reset = true;
+            g_errors |= _BV(RBSM_EID_RC_LOST_SIGNAL);
+            brake_needs_reset = true;
         }
-
-        if((g_is_autonomous) && 
-           (delta4 > CONNECTION_TIMEOUT_US ||
-           delta5 > CONNECTION_TIMEOUT_US))
+        // then check for timeout under autonomous
+        else if((g_is_autonomous == true) &&
```

Comment:
I don't think that this should be an else if statement.  With this, if we time out from RC, then we'll never check to see that high level messages are coming in.  Timing out from RC and high level are independent, so we should be always check both.